build-tools/packages/build-cli/src/commands/generate/changelog.ts

```diff
@@ -0,0 +1,105 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Flags } from "@oclif/core";
+import { command, Options } from "execa";
+import { PackageCommand, PackageKind } from "../../BasePackageCommand";
+import { Repository } from "../../lib";
+import { readFile, writeFile } from "fs/promises";
+import { CleanOptions } from "simple-git";
+import path from "path";
+import { Package } from "@fluidframework/build-tools";
+
+async function replaceInFile(search: string, replace: string, filePath: string): Promise<void> {
+	const content = await readFile(filePath, "utf8");
+	const newContent = content.replace(new RegExp(search, "g"), replace);
+	await writeFile(filePath, newContent, "utf8");
+}
+
+export default class GenerateChangeLogCommand extends PackageCommand<
+	typeof GenerateChangeLogCommand
+> {
+	static description = "Generate a changelog for a new version";
+
+	static flags = {
+		version: Flags.string({
+			description: "The version for which to generate the changelog",
+			required: true,
+		}),
+		path: Flags.directory({
+			description:
+				"Path to a directory containing a package. The version will be loaded from the package.json in this directory.",
+			exists: true,
+			exclusive: ["version"],
+		}),
```

Comment:
This should be the path to the changesets folder. It's not used to load the version, it's used to figure out where the changesets files are.

Reply:
But now I realize I was confused; this whole flag is not needed.